### PR TITLE
disable write permissions primary keys

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/12246

## Testing
**URL to test:** 

local

**Steps to test:**

i made these changes manually by examining the `update` and `insert` permissions for all tables in `tables.yaml` and cross-referencing those with the table's primary key. if the primary key showed up in any of these cases for any role i slashed it.

- afaik the best way to test this is to read through the deletions in `tables.yaml` and confirm that they are all related to `update` or `insert` permissions. alternatively, you can go through the hasura console by clicking through the permissions on each table but i think that would be much more time-consuming

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Review by johnclary — APPROVED** [review]
Wow, it's huge relief to have these permissions tightened up! I ran through a few test cases locally with admin and editor permissions and all seems to be working normally. 

Thank you!!!!!  🙌 🚢 🚢 🚢

**Review by roseeichelmann — APPROVED** [review]
looks good to me! its great this is getting addressed, thanks tilly!

**Review by mddilley — CHANGES_REQUESTED** [review]
This looks awesome, Tilly! So much improvement. ✨ I think that I found one last primary key permission to revoke - insert and update of `project_component_id ` in the `moped_proj_components` table. If I missed something different about that one, I'll turn this right into a ✅.

File: moped-database/metadata/tables.yaml [hunk 1]
```diff
@@ -3941,7 +3905,6 @@
           - added_by
           - interim_project_id
           - parent_project_id
-          - project_id
```
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-database/metadata/tables.yaml:3944]: +1 - the ones without `id` for the primary key are tricky!

**Comment by tillyw** [comment]
> This looks awesome, Tilly! So much improvement. ✨ I think that I found one last primary key permission to revoke - insert and update of `project_component_id ` in the `moped_proj_components` table. If I missed something different about that one, I'll turn this right into a ✅.

that one slipped past me - thanks for catching it! i've pushed a new commit that delete the pkey `update` and `insert` permissions for `moped_proj_components`!

**Review by mddilley — APPROVED** [review]
Looks amazing! We can rest easy knowing that these columns are locked down now 🔒✨

**Review by chiaberry — APPROVED** [review]
ditto to everyone else! we can sleep soundly now

**Pull request merged**